diff --git a/ConfigureChecks.cmake b/ConfigureChecks.cmake
index c82f099..c2064dd 100644
--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -84,9 +84,7 @@ check_function_exists(printf HAVE_PRINTF)
 check_function_exists(setjmp HAVE_SETJMP)
 check_function_exists(signal HAVE_SIGNAL)
 check_function_exists(strsignal HAVE_STRSIGNAL)
-check_function_exists(sprintf HAVE_SNPRINTF)
 check_function_exists(strcmp HAVE_STRCMP)
-check_function_exists(vsnprintf HAVE_VSNPRINTF)
 check_function_exists(clock_gettime HAVE_CLOCK_GETTIME)

 if (WIN32)
@@ -94,6 +92,11 @@ if (WIN32)
     check_function_exists(_vsnprintf HAVE__VSNPRINTF)
     check_function_exists(_snprintf HAVE__SNPRINTF)
     check_function_exists(_snprintf_s HAVE__SNPRINTF_S)
+    check_symbol_exists(snprintf stdio.h HAVE_SNPRINTF)
+    check_symbol_exists(vsnprintf stdio.h HAVE_VSNPRINTF)
+else (WIN32)
+    check_function_exists(sprintf HAVE_SNPRINTF)
+    check_function_exists(vsnprintf HAVE_VSNPRINTF)
 endif (WIN32)

 find_library(RT_LIBRARY rt)
